packages/@aws-cdk/aws-ec2/lib/launch-template.ts

```diff
@@ -0,0 +1,707 @@
+        // Also not implemented in Instance L2
+        // licenseSpecifications: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-metadataoptions
+        // metadataOptions: undefined,
+
+        // https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-tagspecifications
+        // Should be implemented via the Tagging aspect in CDK core. Complication will be that this tagging interface is very unique to LaunchTemplates.
+        // tagSpecification: undefined
+
+        // CDK has no abstraction for Network Interfaces yet.
+        // networkInterfaces: undefined,
+
+        // CDK has no abstraction for Placement yet.
+        // placement: undefined,
+
+      },
+    });
+
+    Tags.of(this).add(NAME_TAG, this.node.path);
+
+    this.defaultVersionNumber = resource.attrDefaultVersionNumber;
+    this.latestVersionNumber = resource.attrLatestVersionNumber;
+    this.launchTemplateId = resource.ref;
+    this.versionNumber = Token.asString(resource.getAtt('LatestVersionNumber'));
+  }
+
+  /**
+   * Allows specifying security group connections for the instance.
+   *
+   * @note Only available if you provide a securityGroup when constructing the LaunchTemplate.
+   */
+  public get connections(): Connections {
+    if (!this._connections) {
+      throw new Error('connections not available on LaunchTemplate. You must provide a securityGroup when constructing the LaunchTemplate to make it available.');
+    }
+    return this._connections;
+  }
+
+  /**
+   * Principal to grant permissions to.
+   *
+   * @note Only available if you provide a role when constructing the LaunchTemplate.
+   */
+  public get grantPrincipal(): iam.IPrincipal {
+    if (!this._role) {
+      throw new Error('grantPrincipal not available on LaunchTemplate. You must provide a role when constructing the LaunchTemplate to make it available.');
+    }
+    return this._role;
+  }
+
+  /**
+   * The type of OS on which the instance is running.
+   *
+   * @note Only available if you provide a machineImage when constructing the LaunchTemplate
+   */
+  public get osType(): OperatingSystemType {
+    // Explicitly check for undef. osType is an enum, and can have the value 0 which messes with attempts at "if (!osType)"
+    if (this._osType === undefined) {
+      throw new Error('osType not available on LaunchTemplate. You must provide a machineImage when constructing the LaunchTemplate to make it available.');
+    }
+    return this._osType;
+  }
+
+  /**
+   * IAM Role assumed by instances that are launched from this template.
+   *
+   * @note Only available if you provide a role when constructing the LaunchTemplate.
+   */
+  public get role(): iam.IRole {
```

Comment:
Since the `role` property isn't really implementing a protocol such as `IGrantable` or `IConnectable`, I'd be fine surfacing the reality here. Declare this just as 

```ts
public readonly role?: iam.IRole;
```

And users can deal with the fact that it might or might not be set.

Reply:
(Because also: think about how `Instance` would consume this.

It needs to do something like:

```
if (props.role) { 
  instanceRole = props.role;
} else if (!launchTemplate.role) {
}